test/python/prefix/wsgi.py

```diff
@@ -0,0 +1,10 @@
+def application(environ, start_response):
+    start_response(
+        '200',
+        [
+            ('Content-Length', '0'),
+            ('Script-Name', environ.get('SCRIPT_NAME', 'NULL')),
+            ('Path-Info', environ['PATH_INFO']),
```

Comment:
I'd like to understand better what is the relation between prefix and SCRIPT_NAME and PATH_INFO.  Could you please explain it to me?

Reply:
From the [WSGI specification](https://peps.python.org/pep-3333/#environ-variables):
> `SCRIPT_NAME`
> The initial portion of the request URL’s “path” that corresponds to the application object, so that the application knows its virtual “location”. This may be an empty string, if the application corresponds to the “root” of the server.

> `PATH_INFO`
> The remainder of the request URL’s “path”, designating the virtual “location” of the request’s target within the application. This may be an empty string, if the request URL targets the application root and does not have a trailing slash.

As an example, say I have a server with a help desk application. The help desk application is the only application, so all HTTP requests to it are routed through that application. A request to `/tickets/20/` will result in the application being called with an empty `SCRIPT_NAME` and a `PATH_INFO` of `/tickets/20/`.

If I someday wish to add a separate customer survey application to the same server, I may update the application server configuration to route requests to `/help` and its descendants to the help desk, and requests to `/survey` and its descendants to the survey application.

In this case, a request to `/help/tickets/20/` will result in the help desk application being called with a `SCRIPT_NAME` of `/help` and a `PATH_INFO` of `/tickets/20/`. The routing logic doesn't need to be changed, and the `SCRIPT_NAME` can be used during server-side HTML rendering so links to `/tickets/all/` are now links to `/help/tickets/all/`.